mono/metadata/image.c

```diff
@@ -2588,14 +2563,12 @@ mono_find_mempool_owner (void *ptr)
 
 		// Iterate over images within a hash
 		g_hash_table_iter_init (&iter, target);
-		while (searching && g_hash_table_iter_next(&iter, NULL, (gpointer *)&image))
+		while (!owner && g_hash_table_iter_next(&iter, NULL, (gpointer *)&image))
 		{
 			mono_image_lock (image);
```

Comment:
Should I do something about this?

The audits are really not going to be possible if we can't acquire locks. As for nesting image_lock inside images_lock, the only way I can think to get around that is to make a copy of the list of all images in the system… every time… which will be a bit performance painful maybe even by the checked build standards.